src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -159,6 +166,91 @@ internal void AddProvider(string workerDir)
         }
 
+        private List<RpcWorkerDescriptionProfile> ReadWorkerDescriptionFromProfiles(JObject workerConfig)
+        {
+            List<RpcWorkerDescriptionProfile> descriptionProfiles = new List<RpcWorkerDescriptionProfile>();
+            JToken profiles = workerConfig.GetValue(RpcWorkerConstants.WorkerDescriptionProfiles);
+            if (profiles != null)
+            {
+                try
+                {
+                    descriptionProfiles = profiles.ToObject<List<RpcWorkerDescriptionProfile>>();
+                }
+                catch (Exception ex)
+                {
+                    _logger?.LogDebug(ex, $"Failed to parse profiles. Default worker description will be used.");
+                }
+            }
+            return descriptionProfiles;
+        }
+
+        private RpcWorkerDescription LoadWorkerDescriptionFromProfiles(List<RpcWorkerDescriptionProfile> descriptionProfiles, RpcWorkerDescription defaultWorkerDescription)
+        {
+            foreach (var profile in descriptionProfiles)
+            {
+                if ( profile.Validate() && CheckProfileConditions(profile))
+                {
+                    RpcWorkerDescription profileDescription = profile.ProfileDescription;
+                    profileDescription.Arguments = profileDescription.Arguments?.Count > 0 ? profileDescription.Arguments : defaultWorkerDescription.Arguments;
+                    profileDescription.DefaultExecutablePath = string.IsNullOrEmpty(profileDescription.DefaultExecutablePath) ? defaultWorkerDescription.DefaultExecutablePath : profileDescription.DefaultExecutablePath;
+                    profileDescription.DefaultWorkerPath = string.IsNullOrEmpty(profileDescription.DefaultWorkerPath) ? defaultWorkerDescription.DefaultWorkerPath : profileDescription.DefaultWorkerPath;
+                    profileDescription.Extensions = profileDescription.Extensions?.Count > 0 ? profileDescription.Extensions : defaultWorkerDescription.Extensions;
+                    profileDescription.Language = string.IsNullOrEmpty(profileDescription.Language) ? defaultWorkerDescription.Language : profileDescription.Language;
+                    profileDescription.WorkerDirectory = string.IsNullOrEmpty(profileDescription.WorkerDirectory) ? defaultWorkerDescription.WorkerDirectory : profileDescription.WorkerDirectory;
+                    return profileDescription;
+                }
+            }
+            return defaultWorkerDescription;
+        }
+
+        private bool CheckProfileConditions(RpcWorkerDescriptionProfile profile)
+        {
+            foreach (var condition in profile.Conditions)
+            {
+                if (!ValidateProfileCondition(condition))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private bool ValidateProfileCondition(ProfileCondition condition)
+        {
+            string value = string.Empty;
```

Comment:
With the condition implementation approach described above, this logic is encapsulated by each condition, removing the responsibility from this type, and the profile evaluation becomes a matter of ensuring all conditions return true from their validation method.